### PR TITLE
DDF-1964 Update Sources Tab to show additional information for registry entries

### PR DESCRIPTION
**Pull request by gordocanchola**
#### What does this PR do?
This PR updates sources tab to show additional information for registry entries. It also adds accordions to the Binding Configurations.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@ryeats @stustison
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@lessarderic
@shaundmorris
#### How should this be tested?
  Verify the build.
#### What are the relevant tickets?
DDF-1964
#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

Adding UI changes
  Adding accordion to the Binding Configurations section
  Adding action provider link in the sources tab

Adding action providers to the allSourceInfo object
Removing updatePublications that was moved to another class in another ticket

### CODE REVIEW

File: catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java [hunk 1]
```diff
@@ -230,32 +207,16 @@ public boolean sourceStatus(String servicePID) {
...
+                        for (String key : Collections.list(properties.keys())) {
+                            plist.put(key, properties.get(key));
                         }
                         source.put(MAP_ENTRY_PROPERTIES, plist);
-
+                        source.put(MAP_ENTRY_REPORT_ACTIONS,
+                                getActions(config, reportActionProviders));
+                        source.put(MAP_ENTRY_OPERATION_ACTIONS,
+                                getActions(config, operationActionProviders));
                         configurations.add(source);
```
Comment (lessarderic) [inline catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: Just curious... `configurations` is a `List<Map<String, Object>>` that contains a `source` which is itself a `Map<String, Object>`, which contains the `MAP_ENTRY_OPERATION_ACTIONS` entry whose value is of type `List<Map<String, String>>`, right? My head hurts :-(

File: catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerTest.java [hunk 2]
```diff
@@ -78,66 +60,30 @@
...
         results.add(new ResultImpl(metacard1));
 
-        filterBuilder = new GeotoolsFilterBuilder();
+        poller = new AdminPollerTest().new MockedAdminPoller(null);
 
-        poller = new AdminPollerTest().new MockedAdminPoller(null,
-                catalogFramework,
-                filterBuilder,
-                catalogStoreMap);
+        ArrayList<ActionProvider> operationActions = new ArrayList<>();
```
Comment (lessarderic) [inline catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerTest.java]: We could use Guava's `ImmutableList.of()` or `Arrays.asList()` to create those lists.

File: catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerTest.java [hunk 3]
```diff
@@ -148,13 +94,20 @@ public void testAllSourceInfo() {
...
 
-        // Assert that the password value was changed from "secret" to "password".
-        String password =
-                (String) ((Map<String, Object>) ((List<Map<String, Object>>) sources.get(1)
-                        .get("configurations")).get(0)
-                        .get("properties")).get("password");
-        assertThat(password, is(equalTo("password")));
+        List<Map<String, Object>> operationActions =
+                (List) configurationMap.get("operation_actions");
+        assertThat(operationActions, hasSize(2));
```
Comment (lessarderic) [inline catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerTest.java]: We should use `hasItems()` instead of `hasSize()` here to make sure the right actions are in the list.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js [hunk 4]
```diff
@@ -14,210 +14,222 @@
...
+                    this.addDisabledConfiguration(currentConfig);
+                }
+                this.set({
+                    currentConfiguration: configuration
+                });
+
+                var pid = configuration.id;
+                var statusModel = new Status.Model(pid);
+                statusModel.on('sync', function () {
+                    wreqr.vent.trigger('status:update-' + pid, statusModel);
```
Comment (lessarderic) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js]: Should we define some "constants" for things like `'sync'`, `'statuc'`, etc.?

**Comment by lessarderic** [comment]
Except for the couple of minor comments, the backend code LGTM 👍 

Did take a cursory look at the front end code but since I'm not that familiar with backbone/marionette, you may want to consider adding @stustison as a reviewer for that part.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js [hunk 5]
```diff
@@ -326,16 +329,16 @@ define(function (require) {
...
-            }
-        ]
+        relations: [{
+            type: Backbone.Many,
+            key: 'value',
+            relatedModel: Service.Model,
+            includeInJSON: false
+        }],
+        parse: function (data) {
+            return data;
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js]: Why did you need to add this function in here? This is the backbone default action anyways.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js [hunk 6]
```diff
@@ -140,77 +159,101 @@ function (ich,Marionette,Backbone,ConfigurationEdit,Service,Utils,wreqr,_,$,moda
...
+            var id = link.attr('id');
+            var failed = $(this.$('#' + id + '-failed')[0]);
+            var success = $(this.$('#' + id + '-success')[0]);
+            var spinner = $(this.$('#' + id + '-spinner')[0]);
+            var httpMethod = this.getHttpMethod(actionId);
+            link.addClass('inactive-link');
+            spinner.show();
+            failed.hide();
+            success.hide();
+            $.ajax({
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: Shouldn't be doing ajax actions from a view. Going back and forth to the server is the responsibility of a model in backbone land.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/templates/sourceModal.handlebars [hunk 7]
```diff
@@ -11,34 +11,72 @@
...
+            <label class="control-label2 container-fluid">Binding Configurations:</label>
+            <div class="modal-accordions container-fluid">
+            </div>
+            <label class="control-label2  container-fluid">Reports:</label>
+            <div class="modal-reports container-fluid">
+                {{#if reportActions}}
+                    <ul class="report-list">
+                        {{#each reportActions}}
+                            <li>
+                                <a href="{{url}}">{{title}}</a>
```
Comment (mcalcote) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/templates/sourceModal.handlebars]: `<a href="{{url}}" target="_blank">{{title}}</a>`
Should make report open in new tab

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js [hunk 8]
```diff
@@ -140,77 +159,101 @@ function (ich,Marionette,Backbone,ConfigurationEdit,Service,Utils,wreqr,_,$,moda
...
         },
+        getActions: function (action) {
+            var disabledConfigs = this.model.get('disabledConfigurations');
+            var currentConfig = this.model.get('currentConfiguration');
+            var actions;
+            if (!_.isUndefined(currentConfig)) {
+                actions = currentConfig.get(action);
+            }
+            if (!_.isUndefined(disabledConfigs)) {
+                var actionConfig = _.find(disabledConfigs.models, function (config) {
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: This isn't the right way to do this. You can just do this.model.get('disabledConfigurations').find(...)

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js [hunk 9]
```diff
@@ -140,77 +159,101 @@ function (ich,Marionette,Backbone,ConfigurationEdit,Service,Utils,wreqr,_,$,moda
...
             }
             if (!_.isUndefined(disabledConfigs)) {
-                disabledConfigs.each(function(config) {
+                disabledConfigs.each(function (config) {
                     configs.add(config.get('service'));
                 });
             }
             return configs;
         },
+        getActions: function (action) {
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: This all seems like business logic that should go into the model.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js [hunk 10]
```diff
@@ -365,25 +414,80 @@ function (ich,Marionette,Backbone,ConfigurationEdit,Service,Utils,wreqr,_,$,moda
                     });
                 }
             }
+
             return config;
         },
+        /**
+         *  Method to get a list of all configs, each with its corresponding service and properties
+         */
+        getAllConfigsWithServices: function () {
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: This seems like business logic. Model?

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js [hunk 11]
```diff
@@ -365,25 +414,80 @@ function (ich,Marionette,Backbone,ConfigurationEdit,Service,Utils,wreqr,_,$,moda
...
+            var configsWithServices = [];
+            listOfConfigStrings.forEach(function (conString) {
+                configsWithServices.push(this.findConfigFromId(conString));
+            }.bind(this));
+            return configsWithServices;
+        },
+        /**
+         *  Helper method for getOrganizationalModel
+         */
+        getOrgProperty: function (orgObj, valueToFind, valueIfNotFound) {
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: Same here, not a view thing.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js (hunk 5 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js]: Removed.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js [hunk 12]
```diff
@@ -14,332 +14,336 @@
...
+                        });
+                        wreqr.vent.trigger('refreshSources');
+                        view.$el.modal("hide");
+                    }).done();
+                }
+            },
+            createDeletePromise: function (source, config) {
+                var deferred = Q.defer();
+                var serviceModels = this.model.get('model').get('value');
+                config.destroy().success(function () {
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: It doesn't look like this promise is chained correctly. `config.destroy()` will get called and then the function will return with the promise from Q, but the destroy promise will happen asynchronously from the Q promise. You could just return `config.destroy` I believe.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js [hunk 13]
```diff
@@ -14,332 +14,336 @@
...
+            initialize: function (options) {
+                this.application = options.application;
+                this.listenTo(wreqr.vent, "showModal", this.showModal);
+            },
+            showModal: function (modalView) {
+                // Global div for workaround with iframe resize and modals
+                var region = this.application.getRegion('sourcesModal');
+                var collectionRegion = this.application.getRegion('collectionRegion');
+                var iFrameModalDOM = $('#IframeModalDOM');
+                modalView.$el.on('hidden.bs.modal', function () {
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: These events are never removed. This is going to turn each modalView into a zombie.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js [hunk 14]
```diff
@@ -14,332 +14,336 @@
...
+                                view.model.get('collection').removeSource(item);
+                                view.model.get('model').get('value').remove(item);
+                            }
+                        });
+                        wreqr.vent.trigger('refreshSources');
+                        view.$el.modal("hide");
+                    }).done();
+                }
+            },
+            createDeletePromise: function (source, config) {
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: This doesn't look like it has anything to do with a view.

File: catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java [hunk 15]
```diff
@@ -230,32 +207,16 @@ public boolean sourceStatus(String servicePID) {
...
-                                            .equals(key) && AttributeDefinition.PASSWORD
-                                            == (Integer) metatypeProperty.get("type")) {
-                                        propertyValue = "password";
-                                        break;
-                                    }
-                                }
-                            }
-                            plist.put(key, propertyValue);
+                        for (String key : Collections.list(properties.keys())) {
+                            plist.put(key, properties.get(key));
```
Comment (stustison) [inline catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: Just do `plist = PropertiesLoader.toMap(properties)`

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Accordion.js [hunk 16]
```diff
@@ -0,0 +1,22 @@
+/**
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Accordion.js]: You don't need this file at all.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Accordion.collection.js [hunk 17]
```diff
@@ -0,0 +1,25 @@
+/**
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Accordion.collection.js]: You don't need this file at all.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js [hunk 18]
```diff
@@ -14,210 +14,222 @@
...
+                    this.set("currentConfiguration", undefined);
+                }
+            },
+            setCurrentConfiguration: function (configuration) {
+                // check to see if the source already has a 'currentConfiguration'
+                // if it does, make the currentConfiguration disabled and move it to the disabledConfiguration
+                // list
+                var currentConfig = this.get('currentConfiguration');
+                if (currentConfig) {
+                    currentConfig.makeDisableCall();
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js]: This isn't a synchronous call. Do you need to be waiting before you add it as a disabled configuration? On a slow connection, it might not actually be disabled.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js [hunk 19]
```diff
@@ -114,19 +109,19 @@ define(function (require) {
...
                     url: url,
                     dataType: 'json'
-                }).done(function(){
+                }).done(function () {
                     model.trigger('disabled');
                     //disabling the model means the PID will be regenerated. This model no longer exists on the server.
                     model.destroy();
-                }).fail(function(){
+                }).fail(function () {
                     new Error('Could not disable configuration ' + pid);
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js]: What are you doing with this error? You need to throw it or something.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ConfigurationEdit.view.js [hunk 20]
```diff
@@ -116,73 +116,103 @@ define([
                 }
             }
         },
-        saveValues: function() {
+        saveValues: function () {
             var values = [];
-            _.each(this.collectionArray.models, function(model) {
+            _.each(this.collectionArray.models, function (model) {
```
Comment (stustison) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ConfigurationEdit.view.js]: You can just do `this.collectionArray.each(function())`

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/templates/sourceModal.handlebars (hunk 7 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/templates/sourceModal.handlebars]: Added the _blank target

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js (hunk 8 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: Changed.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js (hunk 9 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: Moved to the model.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js (hunk 6 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: Moved the ajax call to the model

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js (hunk 10 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: Moved to the model

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js (hunk 11 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js]: Wasn't using getOrgProperty, removed it.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js (hunk 12 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: This is showing up as a change for formatting purposes only but I think the promise chain looks OK. It looks like the caller is waiting for the promise from Q to be resolved. In the destroy's callback it does some clean up and then resolves the promise from Q, returning a couple of things that the caller expects.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js (hunk 13 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: Now Offing the events

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Accordion.js (hunk 16 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Accordion.js]: Removed the file

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Accordion.collection.js (hunk 17 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Accordion.collection.js]: Removed the file

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js (hunk 18 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js]: Removed that logic it was no longer being used.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js (hunk 19 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js]: This was just triggered as changed because of formatting. It doesn't look like anything is being done with that error. The enable call above has the same behavior. Should I write a ticket for this?

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ConfigurationEdit.view.js (hunk 20 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ConfigurationEdit.view.js]: Changed

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js (hunk 14 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: This was only a formatting change should I write a ticket to move this?

Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: Ha ha, just kidding. Moved it

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js (hunk 19 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js]: Removed handling the fail since it wasn't doing anything anyway.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js (hunk 4 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Source.js]: Created Constants

File: catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerTest.java (hunk 2 above)
Comment (gordocanchola) [inline catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerTest.java]: Changed to uns ImmutableList

File: catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java (hunk 1 above)
Comment (gordocanchola) [inline catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: Yes, that's right. Sorry for the headache.

File: catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerTest.java (hunk 3 above)
Comment (gordocanchola) [inline catalog/admin/admin-poller-service/src/test/java/org/codice/ddf/catalog/admin/poller/AdminPollerTest.java]: Changed

File: catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java (hunk 15 above)
Comment (gordocanchola) [inline catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: PropertiesLoader.toMap expects Properties config.getProperties returns a Dictionary

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js [hunk 21]
```diff
@@ -87,47 +81,44 @@ define(function (require) {
...
                 return $.ajax({
                     url: url,
                     dataType: 'json'
-                }).done(function(){
+                }).done(function () {
                     // massage some data to match the new backend pid.
                     model.trigger('enabled');
                     //enabling the model means the PID will be regenerated. This model no longer exists on the server.
                     model.destroy();
-                }).fail(function(){
```
Comment (tbatie) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js]: Shouldn't you be handling the fail case still?

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js [hunk 22]
```diff
@@ -87,47 +81,44 @@ define(function (require) {
...
             if (pid) {
                 return $.ajax({
                     url: url,
                     dataType: 'json'
-                }).done(function(){
+                }).done(function () {
                     model.trigger('disabled');
                     //disabling the model means the PID will be regenerated. This model no longer exists on the server.
                     model.destroy();
-                }).fail(function(){
```
Comment (tbatie) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js]: Same question

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js (hunk 21 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js]: The promise is being returned to the caller so it's available for the caller to handle the fail case. It looks like the fail case was being tapped to log some error. Since it wasn't really doing anything and it doesn't prevent callers from handling the fail case i figure it's safe to remove.

File: catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js (hunk 22 above)
Comment (gordocanchola) [inline catalog/admin/module/catalog-admin-module-sources/src/main/webapp/js/model/Service.js]: Same answer

**Comment by stustison** [comment]
👍

File: catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java [hunk 23]
```diff
@@ -274,114 +235,28 @@ public int compare(Map<String, Object> o1, Map<String, Object> o2) {
...
-                        //The metacard is still in the catalogStore, thus it was persisted
-                        updatedPublishedLocations.add(id);
-                    }
-                }
+    private List<Map<String, String>> getActions(Configuration config,
+            List<ActionProvider> providers) {
+        List<Map<String, String>> actions = new ArrayList<>();
+        for (ActionProvider provider : providers) {
+            if (!provider.canHandle(config)) {
+                continue;
```
Comment (Lambeaux) [inline catalog/admin/admin-poller-service/src/main/java/org/codice/ddf/catalog/admin/poller/AdminPollerServiceBean.java]: Very nice use of a `continue` here.

**Comment by Lambeaux** [comment]
Hero'd 👍

**Comment by shaundmorris** [comment]
The bamboo build was successful on this PR as well.
